### ISSUE TITLE
arrow-return-shorthand inside JSX props

### ISSUE THREAD

**Issue by SaphuA** [palantir/tslint#2191]
### Bug Report

- __TSLint version__: `4.4.2` (coming from `4.3.1`)
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: Webpack

#### TypeScript code being linted

```ts
    return (
        <FilterFileByDropdown valueSelector={(filterBy: FilterFileBy) => { return filterBy.toString(); } } />
    );
```
#### Actual behavior

Retrieving a warning:
`This arrow function body can be simplified by omitting the curly braces and the keyword 'return'. arrow-return-shorthand`

But changing the code to the following makes typescript unhappy:

```ts
    return (
        <FilterFileByDropdown valueSelector={(filterBy: FilterFileBy) => filterBy.toString(); } />
    );
```

#### Expected behavior

No warning.

**Comment by adidahiya** [comment]
What's the TypeScript error the second code snippet produces?

**Comment by ajafff** [comment]
I guess the semicolon is the problem. If you remove it, it should be correct.

**Comment by SaphuA** [comment]
@ajafff I was just coming back to report this. It was indeed the semicolon that was giving the errors.
